### DISCUSSION TITLE
Ethernet with HSPI?

### DISCUSSION THREAD

**Discussion by m8nix** [espressif/arduino-esp32#10274] [Q&A] [answered]
Hello,

is there a way to initialize the Ethernet library on the HSPI-bus (SPI2), like this?
```
#include <SPI.h>
#include <Ethernet.h>

#define SPI_SS 15

SPIClass mySPI(HSPI);

void setup() {
  Ethernet.init(mySPI, SPI_SS);   // -> error: no matching function for call to 'EthernetClass::init(SPIClass&, int)
}
```
The [EthernetSPI2 library](https://github.com/arhi/EthernetSPI2) has not been maintained for a long time and apparently no longer works.

**Comment by JAndrassy** [comment, marked as answer]
why do you want to use the old Ethernet library with the ESP32? in version 3 ESP32 Arduino platform has its own Ethernet library ETH.h which supports any SPI settings. (alternatively you can use my EthernetESP32 library, which is more similar to the Arduino Ethernet library )

**Comment by m8nix** [comment]
> why do you want to use the old Ethernet library with the ESP32? in version 3 ESP32 Arduino platform has its own Ethernet library ETH.h which supports any SPI settings. (alternatively you can use my EthernetESP32 library, which is more similar to the Arduino Ethernet library )

Oops, sorry.
The internet is too full of outdated examples and instructions.
The ETH.h library works great.
Thank you very much.
